src/ast.ts

```diff
@@ -399,40 +425,68 @@ export class Symbols {
       isType: <ExpType> isType,
       methods: methods
     };
-    this.register('paths', '/' + parts.join('/'), p);
+    return <Path> this.register('paths', '/' + parts.join('/'), p);
   }
 
-  registerSchema(name: string, derivedFrom: ExpType | ExpUnionType | void, properties, methods) {
-    methods = methods || {};
+  registerSchema(name: string,
+                 derivedFrom?: ExpType,
+                 properties?: { [prop: string]: ExpType },
+                 methods?: { [name: string]: Method },
+                 params?: string[]): Schema {
     properties = properties || {};
```

Comment:
You don't need these. Just make the parameter non-optional but give it a default value of properties = {}